be/src/exprs/arithmetic_operation.h

```diff
@@ -193,58 +195,123 @@ struct ArithmeticBinaryOperator {
             if constexpr (lt_is_float<Type>) {
-                return b.CreateFAdd(l, r);
+                result.value = b.CreateFAdd(l, r);
             } else {
-                return b.CreateAdd(l, r);
+                result.value = b.CreateAdd(l, r);
             }
         } else if constexpr (is_sub_op<Op>) {
             if constexpr (lt_is_float<Type>) {
-                return b.CreateFSub(l, r);
+                result.value = b.CreateFSub(l, r);
             } else {
-                return b.CreateSub(l, r);
+                result.value = b.CreateSub(l, r);
             }
         } else if constexpr (is_mul_op<Op>) {
             // TODO(Yueyang): implement float type * 0.
             if constexpr (lt_is_float<Type>) {
-                return b.CreateFMul(l, r);
+                result.value = b.CreateFMul(l, r);
             } else {
-                return b.CreateMul(l, r);
+                result.value = b.CreateMul(l, r);
             }
         } else if constexpr (is_div_op<Op>) {
-            // TODO(Yueyang): Support JIT compile of div operator.
-            LOG(WARNING) << "JIT compile of div operator is not supported.";
-            return nullptr;
+            llvm::Value* r_is_zero = nullptr;
+            if constexpr (lt_is_float<Type>) {
+                // return 0 when div by 0.
+                // adjusted_r = r == 0.0 ? r + 1.0 : r;
+                r_is_zero = b.CreateFCmpOEQ(r, llvm::ConstantFP::get(r->getType(), 0));
+                auto* l_is_zero = b.CreateFCmpOEQ(l, llvm::ConstantFP::get(r->getType(), 0));
+                auto* sum = b.CreateFAdd(r, llvm::ConstantFP::get(r->getType(), 1));
+                auto* adjusted_r = b.CreateSelect(r_is_zero, sum, r);
+                result.value = b.CreateFDiv(l, adjusted_r);
+                result.value =
+                        b.CreateSelect(l_is_zero, llvm::ConstantFP::get(r->getType(), 0), b.CreateFDiv(l, adjusted_r));
+            } else {
+                // TODO(Yueyang): avoid 0 div a negative num, make result -0
+                // adjusted_r = r == 0 ? r + 1 : r;
+                r_is_zero = b.CreateICmpEQ(r, llvm::ConstantInt::get(r->getType(), 0));
+                auto* sum = b.CreateAdd(r, llvm::ConstantInt::get(r->getType(), 1));
+                auto* adjusted_r = b.CreateSelect(r_is_zero, sum, r);
+                if constexpr (may_cause_fpe<ResultType> && may_cause_fpe<LType> && may_cause_fpe<RType>) {
+                    // fpe = l == signed_minimum<LType> && r == -1;
+                    auto* cond_left = b.CreateICmpEQ(l, llvm::ConstantInt::get(l->getType(), signed_minimum<LType>));
+                    auto* cond_right = b.CreateICmpEQ(r, llvm::ConstantInt::get(r->getType(), -1));
+                    auto* fpe = b.CreateAnd(cond_left, cond_right);
+                    // modify r to prevent overflow.
+                    adjusted_r = b.CreateSelect(fpe, llvm::ConstantInt::get(l->getType(), 1), adjusted_r);
+                }
```

Comment:
not equal to
```
                if (UNLIKELY(check_fpe_of_min_div_by_minus_one(l, r))) {
                    return signed_minimum<ResultType>;
```

Reply:
The code here is not exactly equivalent, but it should yield the same result. The functionality of the original ﻿if statement is implemented by modifying the divisor. This is done because constructing an ﻿if statement here is difficult, and in a ternary expression, both expressions will be executed, potentially causing a division by zero error. Therefore, the divisor is modified prior to the division operation to ensure correctness.